src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -22,342 +18,201 @@ export const useWithdrawLiquidityHandler = () => {
-      return
-    }
-    if (accountInfo && accountInfo?.address) {
-      if (posManager === null) {
-        return
-      }
-      approveWithdraw(posManager, posManager.address)
-        .then(() => {
-          setEnableRemove(true)
-        })
-        .catch((err: any) => {
-          setEnableRemove(false)
-          notifyError(err)
-        })
-    } else {
-      if (depPool === null) {
-        return
+      if (!gammaPoolContract && GAMMA_POOL_ADDRESS) {
+        setGammaPoolContract(
+          new ethers.Contract(
+            GAMMA_POOL_ADDRESS,
+            GammaPool.abi,
+            accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+          )
+        )
       }
-      approveWithdraw(depPool, depPool.address)
-        .then(() => {
-          setEnableRemove(true)
-        })
-        .catch((err: any) => {
-          setEnableRemove(false)
-          notifyError(err)
-        })
     }
+  }, [provider])
+
+  async function approveTransaction() {
+    let loading = notifyLoading('Waiting for block confirmation')
+    approveWithdraw()
+      .then(() => {
+        setEnableRemove(true)
+        notifyDismiss(loading)
+      })
+      .catch((err) => {
+        notifyDismiss(loading)
+        setEnableRemove(false)
+        notifyError(err.toString())
+      })
   }
 
   async function withdrawLiquidity(balance: number) {
+    if (balance === 0) {
+      throw new Error('Please select a valid amount to withdraw')
+    }
     let amt = '0'
     if (balance === 100) {
       amt = totalLiquidityAmt.toString()
     } else {
       amt = ethers.utils.parseEther(((liquidityAmt * balance) / 100).toString()).toString()
     }
-    // Position Manager contract address
-    let address =
-      process.env.NEXT_PUBLIC_ENVIRONMENT === 'local'
-        ? process.env.NEXT_PUBLIC_POSITION_MANAGER_ADDRESS || ''
-        : DEPOSIT_POOL_ADDRESS
-    if (provider) {
-      if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-        if (accountInfo && accountInfo?.address) {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider.getSigner(accountInfo?.address)))
-        } else {
-          setPosManager(new ethers.Contract(address, PosManager.abi, provider))
-        }
-      } else {
-        if (accountInfo && accountInfo?.address) {
-          setDepPool(new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo?.address)))
-        } else {
-          setDepPool(new ethers.Contract(address, DepPool.abi, provider))
-        }
-      }
-    } else {
-      notifyError('Please connect wallet')
-    }
-
     if (!accountInfo || !accountInfo.address) {
       notifyError('Wallet not connected.')
       return
     }
 
-    if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-      if (posManager === null) {
-        return
-      }
+    if (parseInt(amt) === 0) {
+      throw new Error('User does not have enough reserves to withdraw')
```

Comment:
I think should be notifyError

Reply:
All the throw statements are going to be shown in notifyError in the main method from where it was called